.github/workflows/build.yml

```diff
@@ -0,0 +1,180 @@
+      run: make run-integration-tests
+    - run: docker images
+    - name: Save images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: make save-images
+    - name: Upload images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: images
+        path: images.tar
+        retention-days: 1
+
+  push_images:
+    name: Push images
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_HOSTNAME: "localhost:5000"
+      DOCKER_NAMESPACE: m4d-system
+    needs: [verify, test, integration-tests]
+    if: ${{ github.event_name != 'pull_request' }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Download artifact
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/download-artifact@v2
+      with:
+        name: images
+    - name: Load images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: docker load -i images.tar
+    - run: docker images
+    - name: Publish images
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      run: sudo chmod o+x /etc/docker && make docker-retag-and-push-public && make helm-push-public
+
+  build_website:
+    name: Build website
+    runs-on: ubuntu-latest
+    env:
+      WEBSITE_LOCAL_DIR: public
+    steps:
+    - name: Install asciidoctor
+      run: sudo apt-get -y install asciidoctor
+    - uses: actions/checkout@v2
+    - name: Cache binaries
+      uses: actions/cache@v2
+      with:
+        path: |
+          ~/go/pkg/mod
+          ${{ github.workspace }}/hack/tools/bin
+        key: ${{ runner.os }}-web-${{ hashFiles('hack/tools/bin/*') }}
+        restore-keys: ${{ runner.os }}-web
+    - name: Create website
+      run: make -C website
+    - name: Upload website artifact
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/upload-artifact@v2
+      with:
+        name: website
+        path: website/${{ env.WEBSITE_LOCAL_DIR }}
+        retention-days: 1
+
+  publish_website:
+    name: Publish website
+    runs-on: ubuntu-latest
+    env:
+      WEBSITE_LOCAL_DIR: public
+    needs: [verify, test, integration-tests, build_website]
+    if: ${{ github.event_name != 'pull_request' }}
+    steps:
+    - uses: actions/checkout@v2
+    - name: Download website artifact
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: actions/download-artifact@v2
+      with:
+        name: website
+    - name: Publish website
+      if: ${{ github.event_name != 'pull_request' && env.ACTIVE_CI == 'gh_actions' }}
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: website/${{ env.WEBSITE_LOCAL_DIR }}
+        publish_branch: gh-pages
```

Comment:
[FUTURE]

I'll make a PR to move this to separate workflows with the new website.